ecpy/tasks/tasks/validators.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright 2015 by Ecpy Authors, see AUTHORS for more details.
+#
+# Distributed under the terms of the BSD license.
+#
+# The full license is in the file LICENCE, distributed with this software.
+# -----------------------------------------------------------------------------
+"""Validators for feval members.
+
+"""
+from __future__ import (division, unicode_literals, print_function,
+                        absolute_import)
+
+from traceback import format_exc
+
+from atom.api import Atom, Value, Bool
+
+
+class Feval(Atom):
+    """Object hanlding the validation of feval tagged member.
+
+    """
+    #: Allowed types for the result of the evaluation of the member.
+    types = Value()
+
+    #: Should the validator propagate an error or simply warn the user.
+    warn = Bool()
+
+    def check(self, task_or_interface, member):
+        """Validate the feval formula.
+
+        """
+        str_value = getattr(task_or_interface, member)
+        if not self.should_test(task_or_interface, str_value):
+            return None, True, ''
+
+        msg = ''
+        val = None
+        try:
+
+            task = (task_or_interface.task
+                    if task_or_interface.dep_type == 'ecpy.tasks.interface'
+                    else task_or_interface)
+            val = task.format_and_eval_string(str_value)
+        except Exception:
+            msg = 'Failed to eval %s : %s' % (member, format_exc())
+
+        if not msg:
+            val, msg = self.validate(task_or_interface, val)
+
+        if not msg or (msg and self.warn):
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [The if statement can be reduced by assigning bool of test](https://www.codacy.com/app/marul/ecpy/file/2564166147/issues/source?bid=3033427&fileBranchId=3294218#l52)